src/normalization/Normalizer.cpp

```diff
@@ -249,6 +254,16 @@ void Normalizer::normalize(Module& module) const
         PROFILE_COUNTER_WITH_PREV(vc4c::profiler::COUNTER_NORMALIZATION + 2, "Eliminate Phi-nodes (after)",
             method->countInstructions(), vc4c::profiler::COUNTER_NORMALIZATION + 1);
     }
+
+    {
+        // TODO: move this optimization to appropriate location
+        auto kernels = module.getKernels();
+        for(Method* kernelFunc : kernels)
+        {
+            optimizations::combineDMALoads(module, *kernelFunc, config);
```

Comment:
Does this code have to be run before the normalization steps (e.g. before the memory accesses are rewritten)?